### PR TITLE
utime.h: increase sec type from __u32 to time_t

### PR DESCRIPTION
**Pull request by kungf**
when i set osd_scrub_min(max)_interval to a very large num, just like 6307200000(200*365*24*3600), osd still do scrub. finally i find that the utime_t::nsec only __u32, it's max num is 4294967295, and <
6307200000, so scrub sched_time will overflow, and it will be a random time.

i am confused that why utime_t::sec(nsec) uses type of__u32? it looks like that __u32 is not enough for time desc, why utime_t not like struct timespec:
struct timespec
{
      time_t tv_sec; /*second*/
      long tv_nsec;/*nanosecond*/
}
also, it has to much types  translation in utime.h, like  from __u32 to long, from double to __u32, and so on. it's easy to make a mistake, like i encountered overflow.

i push this pr, and want to get a further learn. thanks @liewegas  

Signed-off-by: kungf <yang.wang@easystack.cn>

### CODE REVIEW

**Comment by kungf** [comment]
@xiexingguo need a review, please

File: src/include/utime.h [hunk 1]
```diff
@@ -122,15 +123,15 @@ class utime_t {
...
+    bl.append((char *)(this), sizeof(time_t) + sizeof(long));
 #else
     ::encode(tv.tv_sec, bl);
     ::encode(tv.tv_nsec, bl);
 #endif
   }
   void decode(bufferlist::iterator &p) {
 #if defined(CEPH_LITTLE_ENDIAN)
-    p.copy(sizeof(__u32) + sizeof(__u32), (char *)(this));
+    p.copy(sizeof(time_t) + sizeof(long), (char *)(this));
```
Comment (cbodley) [inline src/include/utime.h]: changing the encoded format here without any kind of versioning support would make it impossible to upgrade from luminous to mimic without corrupting existing data

File: src/include/utime.h [hunk 2]
```diff
@@ -38,23 +38,24 @@
 class utime_t {
 public:
   struct {
-    __u32 tv_sec, tv_nsec;
+    time_t tv_sec;
+    long tv_nsec;
```
Comment (liewegas) [inline src/include/utime.h]: we can't change the on disk encoding without breaking compatibility

**Comment by liewegas** [comment]
It seems like hte problem is just hte constructor.  utime_t can hold up to 136 years.. if someone sets a time larger than that we should cap it (instead of wrapping).

File: src/include/utime.h (hunk 2 above)
Comment (xiexingguo) [reply src/include/utime.h]: +1

**Comment by kungf** [comment]
i fix the scrub  sched_time overflow in another way, see #21113.

**Pull request closed without merging**